docs(misconf): Add guide on input schema (#3692)
aquasecurity/trivy: docs/docs/misconfiguration/custom/schema.md

# Input Schema

## Overview

> Policies can be defined with custom schemas that allow inputs to be verified against it. Adding an policy schema

Reviewer: ```suggestion
Policies can be defined with custom schemas that allow inputs to be verified against them. Adding a policy schema
```

enables Trivy to show more detailed error messages when an invalid input is encountered.

> For instance if we have a policy that checks for misconfigurations in a `Dockerfile`, we could define the

Reviewer: Dockerfile is a great example for input schemas because dockerfiles are not json compatible by nature and without json schema it is impossible to know what you are even evaluating. I think it's worth mentioning and even reframing the intro to emphasize this.

Author: added more context to show the benefit of input schemas

# Input Schema

## Overview
Policies can be defined with custom schemas that allow inputs to be verified against them. Adding a policy schema
enables Trivy to show more detailed error messages when an invalid input is encountered.

In Trivy we have been able to define a schema for a [Dockerfile](https://github.com/aquasecurity/defsec/blob/master/pkg/rego/schemas/dockerfile.json).
Without input schemas, a policy would be as follows:

!!! example
    ```
    # METADATA
    package mypackage

    deny {
        input.evil == "foo bar"
    }
    ```

If this policy is run against offending Dockerfile(s), there will not be any issues as the policy will fail to evaluate.
Although the policy's failure to evaluate is legitimate, this should not result in a positive result for the scan.

For instance if we have a policy that checks for misconfigurations in a `Dockerfile`, we could define the
schema as such

!!! example
    ```
    # METADATA
    # schemas:
    # - input: schema["dockerfile"]

>     package mypackage

Reviewer 2: why is it called `mypackage`?

Author: Just a name.
